skrub/tests/test_agg_joiner.py

```diff
@@ -0,0 +1,429 @@
+    )
+    agg_join = AggJoiner(
+        tables=[
+            (main, "userId", "rating", "hello"),
+        ],
+        main_key="userId",
+    )
+    with pytest.raises(ValueError, match=msg):
+        agg_join.fit(main)
+
+    # check non dataframe
+    msg = (
+        "'tables' must be a list of tuple and the first element of each "
+        "tuple must be a DataFrame, got <class 'numpy.ndarray'> at index 0."
+    )
+    agg_join = AggJoiner(
+        tables=[
+            (main.values, "userId"),
+        ],
+        main_key="userId",
+    )
+    with pytest.raises(TypeError, match=msg):
+        agg_join.fit(main)
+
+
+def test_X_string_placeholder():
+    msg = "If the dataframe is declared with a string, it can only be 'X', got 'Y'."
+    agg_join = AggJoiner(
+        tables=[
+            ("Y", "userId"),
+        ],
+        main_key="userId",
+    )
+    with pytest.raises(ValueError, match=msg):
+        agg_join.fit(main)
+
+
+y = pd.DataFrame(dict(rating=[4.0, 4.0, 4.0, 3.0, 2.0, 4.0]))
+
+
+@pytest.mark.parametrize(
+    "y, col_name",
+    [
+        (y, "rating"),
+        (y["rating"], "rating"),
+        (y.values, "y0"),
+        (y.values.tolist(), "y0"),
+    ],
+)
+def test_agg_target(y, col_name):
+    agg_target = AggTarget(
+        main_key=["userId"],
+        suffixes=["_user"],
+        agg_ops=["hist(2)", "value_counts"],
+    )
+    main_transformed = agg_target.fit_transform(main, y)
+
+    main_transformed_expected = pd.DataFrame(
+        {
+            "userId": [1, 1, 1, 2, 2, 2],
+            "movieId": [1, 3, 6, 318, 6, 1704],
+            "rating": [4.0, 4.0, 4.0, 3.0, 2.0, 4.0],
+            "genre": ["drama", "drama", "comedy", "sf", "comedy", "sf"],
+            f"{col_name}_(1.999, 3.0]_user": [0, 0, 0, 2, 2, 2],
+            f"{col_name}_(3.0, 4.0]_user": [3, 3, 3, 1, 1, 1],
+            f"{col_name}_2.0_user": [0.0, 0.0, 0.0, 1.0, 1.0, 1.0],
+            f"{col_name}_3.0_user": [0.0, 0.0, 0.0, 1.0, 1.0, 1.0],
+            f"{col_name}_4.0_user": [3.0, 3.0, 3.0, 1.0, 1.0, 1.0],
+        }
+    )
+
+    assert_frame_equal(main_transformed, main_transformed_expected)
+
+
+def test_agg_t_missing_agg_ops():
+    agg_target = AggTarget(
+        main_key=["userId"],
+        suffixes=["_user"],
+    )
+
+    # y is continuous
+    y = pd.DataFrame(dict(rating=[4.0, 4.1, 4.2, 4.3, 4.4, 4.5]))
+    agg_target.fit(main, y)
+    assert agg_target.agg_ops_ == ["mean"]
+
+    # y is categorical
+    y = pd.DataFrame(dict(rating=["1", "2", "3", "1", "2", "3"]))
+    agg_target.fit(main, y)
+    assert agg_target.agg_ops_ == ["mode"]
+
+
+def test_agg_t_check_input():
+    agg_target = AggTarget(
+        main_key=["userId"],
+        suffixes=["_user"],
+    )
+    msg = (
+        "Only Pandas or Polars dataframes are currently supported, "
+        "got [<class 'numpy.ndarray'>]."
+    )
+    with pytest.raises(TypeError, match=re.escape(msg)):
```

Comment:
When testing for error messages, it is often better to check for specific keywords instead of the full sentence.
This is because if it slightly changes (typo, rewording), the test doesn't pass anymore.
For this specific case, it could be:
```suggestion
    # First keyword checks it's an issue of "supported",
    # second one checks that the type of the passed object
    # is explicitly mentioned in the error, which makes it more
    # descriptive, thus easier to debug.
    with pytest.raises(TypeError, match=r"supported|ndarray"):
```
Note: I did not check if it runs

Reply:
That's a great remark, thanks! FYI, the correct version is `r"(?=.*supported)(?=.*ndarray)"`. See [this guide](https://www.ocpsoft.org/tutorials/regular-expressions/password-regular-expression/).